components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/form-builder/form-utils.js

```diff
@@ -434,71 +961,428 @@ define(['require', 'lodash', 'appData', 'log', 'constants', 'handlebar', 'annota
+                                self.configurationData.rawExtensions["windowFunctionNames"])
+                        } else {
+                            predefinedParameters = self.getPredefinedParameters(streamHandlerContent,
+                                self.configurationData.rawExtensions["streamFunctions"])
+                        }
+                        if (self.validateParameters(streamHandlerContent, predefinedParameters)) {
+                            isErrorOccurred = true;
+                            return isErrorOccurred
+                        }
+                    }
+                });
+            }
+            return isErrorOccurred
+        };
+
+        /**
+         * @function to obtain the predefined parameters of particular window or stream-function type
+         */
+        FormUtils.prototype.getPredefinedParameters = function (div, predefinedFunctions) {
+            var predefinedParameters = [];
+            var functionName = $(div).find('.defineFunctionName select').val().toLowerCase()
+            _.forEach(predefinedFunctions, function (predefinedFunction) {
+                if (functionName == predefinedFunction.name.toLowerCase()) {
+                    predefinedParameters = predefinedFunction.parameters;
+                    return false;
+                }
+            });
+            return predefinedParameters;
+        };
+
+        /**
+         * @function for generic validation of parameter values
+         * @param {Object} predefinedParameters predefined parameters of the selected window type
+         * @return {boolean} isError
+         */
+        FormUtils.prototype.validateParameters = function (parameterDiv, predefinedParameters) {
+            var self = this;
+            var isError = false;
+            $(parameterDiv).find('.parameter').each(function () {
+                var parameterValue = $(this).find('.parameter-value').val().trim();
+                var parameterName = $(this).find('.parameter-name').text().trim();;
+                var predefinedParameter = self.getObject(parameterName, predefinedParameters);
+                if (!predefinedParameter.optional) {
+                    if (!self.checkParameterValue(parameterValue, predefinedParameter, this)) {
+                        isError = true;
+                        return false;
+                    }
+                } else {
+                    if ($(this).find('.parameter-checkbox').is(":checked")) {
+                        if (!self.checkParameterValue(parameterValue, predefinedParameter, this)) {
+                            isError = true;
+                            return false;
+                        }
+                    }
+                }
+            });
+            return isError;
+        };
+
+        /**
+         * @function to show autocomplete drop down
+         */
+        FormUtils.prototype.showDropDown = function () {
+
```

Comment:
unnessary space

Reply:
Fixed in 31afee1947aeece2a209055e86b215aee89eb33c